TST: CI is failing on Windows

<!-- This comments are hidden when you submit the issue,
so you do not need to remove them! -->

<!-- Please be sure to check out our contributing guidelines,
https://github.com/astropy/astropy/blob/master/CONTRIBUTING.md .
Please be sure to check out our code of conduct,
https://github.com/astropy/astropy/blob/master/CODE_OF_CONDUCT.md . -->

<!-- Please have a search on our GitHub repository to see if a similar
issue has already been posted.
If a similar issue is closed, have a quick look to see if you are satisfied
by the resolution.
If not please go ahead and open an issue! -->

<!-- Please check that the development version still produces the same bug.
You can install development version with
pip install git+https://github.com/astropy/astropy
command. -->

### Description
<!-- Provide a general description of the bug. -->

Windows job started failing after merging #9929 even though all jobs passed in the PR. Maybe PR was not rebased against latest `master`? Maybe we have corrupted cache? Not sure at this point.

### Expected behavior
<!-- What did you expect to happen. -->

Windows job should pass.

### Actual behavior
<!-- What actually happened. -->
<!-- Was the output confusing or poorly described? -->

It is failing. Example log: https://travis-ci.org/github/astropy/astropy/jobs/672200286

```
______________________ test_single_line_string[False-|] _______________________
delimiter = '|', fast_reader = False
    @pytest.mark.parametrize('delimiter', [' ', '|', '\n', '\r'])
    @pytest.mark.parametrize('fast_reader', [False, True, 'force'])
    def test_single_line_string(delimiter, fast_reader):
        """
        String input without a newline character is interpreted as filename,
        unless element of an iterable. Maybe not logical, but test that it is
        at least treated consistently.
        """
        expected = Table([[1], [2], [3.00]], names=('col1', 'col2', 'col3'))
        text = "1{0:s}2{0:s}3.0".format(delimiter)
    
        if delimiter in ('\r', '\n'):
            t1 = ascii.read(text, format='no_header', delimiter=delimiter, fast_reader=fast_reader)
            assert_table_equal(t1, expected)
        else:
            with pytest.raises(FileNotFoundError):
>               t1 = ascii.read(text, format='no_header', delimiter=delimiter, fast_reader=fast_reader)
..\..\.tox\py37-test-alldeps\lib\site-packages\astropy\io\ascii\tests\test_c_reader.py:1541: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
..\..\.tox\py37-test-alldeps\lib\site-packages\astropy\io\ascii\ui.py:340: in read
    dat = _guess(table, new_kwargs, format, fast_reader)
..\..\.tox\py37-test-alldeps\lib\site-packages\astropy\io\ascii\ui.py:500: in _guess
    dat = reader.read(table)
..\..\.tox\py37-test-alldeps\lib\site-packages\astropy\io\ascii\core.py:1231: in read
    self.lines = self.inputter.get_lines(table, newline=newline)
..\..\.tox\py37-test-alldeps\lib\site-packages\astropy\io\ascii\core.py:306: in get_lines
    encoding=self.encoding) as fileobj:
c:\tools\miniconda3\envs\test\lib\contextlib.py:112: in __enter__
    return next(self.gen)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
name_or_obj = '1|2|3.0', encoding = None, cache = False, show_progress = True
remote_timeout = 10.0, sources = None, http_headers = None
    @contextlib.contextmanager
    def get_readable_fileobj(name_or_obj, encoding=None, cache=False,
                             show_progress=True, remote_timeout=None,
                             sources=None, http_headers=None):

<docstring snipped>
    
        # close_fds is a list of file handles created by this function
        # that need to be closed.  We don't want to always just close the
        # returned file handle, because it may simply be the file handle
        # passed in.  In that case it is not the responsibility of this
        # function to close it: doing so could result in a "double close"
        # and an "invalid file descriptor" exception.
        PATH_TYPES = (str, pathlib.Path)
    
        close_fds = []
        delete_fds = []
    
        if remote_timeout is None:
            # use configfile default
            remote_timeout = conf.remote_timeout
    
        # Get a file object to the content
        if isinstance(name_or_obj, PATH_TYPES):
            # name_or_obj could be a Path object if pathlib is available
            name_or_obj = str(name_or_obj)
    
            is_url = _is_url(name_or_obj)
            if is_url:
                name_or_obj = download_file(
                    name_or_obj, cache=cache, show_progress=show_progress,
                    timeout=remote_timeout, sources=sources,
                    http_headers=http_headers)
>           fileobj = io.FileIO(name_or_obj, 'r')
E           OSError: [Errno 22] Invalid argument: '1|2|3.0'
..\..\.tox\py37-test-alldeps\lib\site-packages\astropy\utils\data.py:237: OSError
_______________________ test_single_line_string[True-|] _______________________
delimiter = '|', fast_reader = True
    @pytest.mark.parametrize('delimiter', [' ', '|', '\n', '\r'])
    @pytest.mark.parametrize('fast_reader', [False, True, 'force'])
    def test_single_line_string(delimiter, fast_reader):
        """
        String input without a newline character is interpreted as filename,
        unless element of an iterable. Maybe not logical, but test that it is
        at least treated consistently.
        """
        expected = Table([[1], [2], [3.00]], names=('col1', 'col2', 'col3'))
        text = "1{0:s}2{0:s}3.0".format(delimiter)
    
        if delimiter in ('\r', '\n'):
            t1 = ascii.read(text, format='no_header', delimiter=delimiter, fast_reader=fast_reader)
            assert_table_equal(t1, expected)
        else:
            with pytest.raises(FileNotFoundError):
>               t1 = ascii.read(text, format='no_header', delimiter=delimiter, fast_reader=fast_reader)
..\..\.tox\py37-test-alldeps\lib\site-packages\astropy\io\ascii\tests\test_c_reader.py:1541: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
..\..\.tox\py37-test-alldeps\lib\site-packages\astropy\io\ascii\ui.py:340: in read
    dat = _guess(table, new_kwargs, format, fast_reader)
..\..\.tox\py37-test-alldeps\lib\site-packages\astropy\io\ascii\ui.py:500: in _guess
    dat = reader.read(table)
..\..\.tox\py37-test-alldeps\lib\site-packages\astropy\io\ascii\fastbasic.py:117: in read
    **self.kwargs)
astropy\io\ascii\cparser.pyx:236: in astropy.io.ascii.cparser.CParser.__cinit__
    ???
astropy\io\ascii\cparser.pyx:278: in astropy.io.ascii.cparser.CParser.setup_tokenizer
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
>   ???
E   OSError: [Errno 22] Invalid argument: '1|2|3.0'
astropy\io\ascii\cparser.pyx:125: OSError
______________________ test_single_line_string[force-|] _______________________
delimiter = '|', fast_reader = 'force'
    @pytest.mark.parametrize('delimiter', [' ', '|', '\n', '\r'])
    @pytest.mark.parametrize('fast_reader', [False, True, 'force'])
    def test_single_line_string(delimiter, fast_reader):
        """
        String input without a newline character is interpreted as filename,
        unless element of an iterable. Maybe not logical, but test that it is
        at least treated consistently.
        """
        expected = Table([[1], [2], [3.00]], names=('col1', 'col2', 'col3'))
        text = "1{0:s}2{0:s}3.0".format(delimiter)
    
        if delimiter in ('\r', '\n'):
            t1 = ascii.read(text, format='no_header', delimiter=delimiter, fast_reader=fast_reader)
            assert_table_equal(t1, expected)
        else:
            with pytest.raises(FileNotFoundError):
>               t1 = ascii.read(text, format='no_header', delimiter=delimiter, fast_reader=fast_reader)
..\..\.tox\py37-test-alldeps\lib\site-packages\astropy\io\ascii\tests\test_c_reader.py:1541: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
..\..\.tox\py37-test-alldeps\lib\site-packages\astropy\io\ascii\ui.py:357: in read
    dat = fast_reader_rdr.read(table)
..\..\.tox\py37-test-alldeps\lib\site-packages\astropy\io\ascii\fastbasic.py:117: in read
    **self.kwargs)
astropy\io\ascii\cparser.pyx:236: in astropy.io.ascii.cparser.CParser.__cinit__
    ???
astropy\io\ascii\cparser.pyx:278: in astropy.io.ascii.cparser.CParser.setup_tokenizer
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
>   ???
E   OSError: [Errno 22] Invalid argument: '1|2|3.0'
astropy\io\ascii\cparser.pyx:125: OSError
==== 3 failed, 14467 passed, 385 skipped, 97 xfailed in 416.70s (0:06:56) =====
Exception ignored in: <function _Database.close at 0x0000019A0B004828>
Traceback (most recent call last):
  File "c:\tools\miniconda3\envs\test\lib\dbm\dumb.py", line 284, in close
    self._commit()
  File "c:\tools\miniconda3\envs\test\lib\dbm\dumb.py", line 135, in _commit
    with self._io.open(self._dirfile, 'w', encoding="Latin-1") as f:
FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\travis\\AppData\\Local\\Temp\\pytest-of-unknown\\pytest-0\\test_lock_behaviour_if_directo0\\astropy\\download\\py3\\urlmap.dir'
ERROR: InvocationError for command 'C:\Users\travis\build\astropy\astropy\.tox\py37-test-alldeps\Scripts\pytest.EXE' --pyargs astropy 'C:\Users\travis\build\astropy\astropy/docs' (exited with code 1)
```

--- a/astropy/io/ascii/tests/test_c_reader.py
+++ b/astropy/io/ascii/tests/test_c_reader.py
@@ -1537,7 +1537,8 @@ def test_single_line_string(delimiter, fast_reader):
         t1 = ascii.read(text, format='no_header', delimiter=delimiter, fast_reader=fast_reader)
         assert_table_equal(t1, expected)
     else:
-        with pytest.raises(FileNotFoundError):
+        # Windows raises OSError, but not the other OSes.
+        with pytest.raises((FileNotFoundError, OSError)):
             t1 = ascii.read(text, format='no_header', delimiter=delimiter, fast_reader=fast_reader)
 
     t2 = ascii.read([text], format='no_header', delimiter=delimiter, fast_reader=fast_reader)

